tools/codestyle/uncrustify_run.sh

```diff
@@ -1,15 +1,37 @@
 #!/usr/bin/env bash
 
-DIR="$1"
+# Go to repository root directory regardless of where script was run from:
+cd "${BASH_SOURCE%/*}/../.."
 
-SRC_DIRS="aeolus audio audiofile avsomr awl bww2mxml crashreporter effects fonttools global importexport \
-libmscore main miditools mscore mtest omr telemetry"
+HERE="tools/codestyle" # path to dir that contains this script
```

Comment:
Maybe `$(dirname $0)`

Reply:
No, not after the `cd` in the [line above](https://github.com/musescore/MuseScore/pull/6329/files#diff-db32b7a1583714d754a939e3de93a648R4). That line is effectively:

```bash
cd "$(dirname "$0")/../.."
```

which puts us in the MuseScore root folder. From there, `HERE` is "tools/codestyle".

Reply:
Or `${BASH_SOURCE%/*}`?

Reply:
That is the same as `"$(dirname "$0")"` but uses Bash built-ins (faster) instead of calling `dirname` which is an external binary (slower).

Basically, `HERE` is how to get from `PWD` to the directory that contains the script. In the other scripts it is simply:

```bash
HERE="$(dirname "$0")" # cfg is at ${HERE}/uncrustify_musescore.cfg
```

But in this file we do `cd` which changes the value of `PWD`:

```bash
HERE="$(dirname "$0")" # cfg is at ${HERE}/uncrustify_musescore.cfg
cd "${HERE}/../.." # PWD has changed so must update HERE
 # cfg no longer at ${HERE}/uncrustify_musescore.cfg
HERE="tools/codestyle" # cfg is at ${HERE}/uncrustify_musescore.cfg
```